tutorials/plot_artifacts_correction_ica.py

```diff
@@ -60,10 +60,14 @@
 # Fit ICA
 # -------
 #
-# ICA parameters:
+# First, choose the ICA method. There are currently 4 possible choices:
+# `fastica`, `picard`, `infomax` and `extended-infomax`. Picard converges
+# faster than the other algorithms, and tends to give more consistent results.
 
+method = 'fastica'
```

Comment:
I'd rather change this to `picard`

Reply:
I thought this PR was about adding Picard to MNE, not making it the new default? Don't get me wrong, but IMO there should be much more testing (ideally also from many different users) happening before it gets promoted to the new default.

Reply:
The change is somewhat separable from the default argument. Especially if it's faster it would be nice to have here. (Or more stable with the same speed.) As long as there is a comment as to why it's chosen for this example I think it's fine to change it here.

Reply:
I agree with @jona-sassenhagen, even if it's just for this example it's probably a bad idea to pull in extra dependencies if there is no clear benefit.

Reply:
`pip`-installable (and in the MNE env) dependencies are pretty straightforward, no? One benefit (in addition to any speed or stability gains, which seem to have been argued already as potential advantages to this algorithm) is a potential increase in users who will try it and give us feedback, which sounds like a necessary feature for people to gain confidence in it..

Reply:
Right, this is not a real issue if it's just for this example. Still, can we quantify the improvement in this example? I.e., how much faster is Picard? How much more stable? Otherwise we're back to belief :smile:...

Reply:
Based on previous comments, I'd expect the same speed but more reliable, or the same reliability and half the speed. We can let @pierreablin show us if he is motivated to win you over :)

Reply:
Argh I mean half the time double the speed

Reply:
I'm thinking a comment like this:
```
# Here we use the 'picard' solver with a permissive tolerance because ICA converges
# very quickly when finding artifacts, and 'picard' does this faster than other variants.
# For a full data decomposition, the default tolerance (1e-4) achieves roughly the same
# speed with higher reliability.
```
Would that win you over @cbrnr ?

Reply:
Yes, you can win me over with hard facts 😄. Although I doubt that this is a suitable example, because @pierreablin has mentioned that artifacts are always among the largest components, so even FastICA should consistently find them.

Reply:
I'll give it a try :)

First, look at this example (inspired by the mne tutorials): https://pierreablin.github.io/picard/auto_examples/plot_ecg_detection.html#sphx-glr-auto-examples-plot-ecg-detection-py
This is simply using ICA to find ECG artifacts. First, I use FastICA with default tol, and repeat the experiment 3 times. You can see that I get two consistent sources, for which the topography clearly matches an ECG. Then, there is a third one for two runs, which is absolutely not consistent and looks like crap, but is still marked as an ECG.

Using Picard, you always get the same ECG sources, and no additional noisy channel. On those data with the given tol/maxiter parameters, picard is also more than two times faster.


But this is only for one subject on a specific task, for a more consistent study, just look at the last figure of our paper  https://arxiv.org/pdf/1711.10873.pdf :  using the same criterion to monitor convergence, you can see that Picard ends up being much faster than Fastica, especially when it comes to solving the problem accurately (by accurately I mean with a small tolerance).

I hope that this convinces you that Picard can be useful. Now I don't advocate for using it as default anywhere, since as said before it has not been thoroughly tested by anyone other than myself.

Reply:
I never said Picard isn't useful. Actually I think it is great that it will be available in MNE! I'm just arguing that if we use it in an existing example, it should perform better in this particular example.

Now you clearly show this in your own example, although the argumentation doesn't fully convince me, because it is based on a ECG detection algorithm and not on the ICA algorithm per se. I'm not an expert in ECG detection, but the "crap" component doesn't actually look like crap - can you rule out that this is not ECG-related? What does convince me more is the fact that FastICA did not converge, therefore its components are likely to be flawed, whereas Picard apparently did converge, Picard is faster, and thus its components are better.

Getting back to the example, we're talking about an MNE example here. I think it would be important to show that Picard also performs better than FastICA in this example (showing that with the default settings FastICA doesn't converge but Picard does would be sufficient IMO). This doesn't seem to be the case, or at least there don't seem to be any problems with FastICA (although further down ECG artifacts are scored, and this yields three components, including one component you'd mark as crap).

In conclusion, if you think Picard performs better than FastICA in the MNE example, go ahead and replace FastICA! If this is not so clear like I think (but maybe I'm wrong), we should carefully consider why we'd want to replace something that does the job.

Reply:
> if you think Picard performs better than FastICA in the MNE example, go ahead and replace FastICA! If this is not so clear like I think (but maybe I'm wrong), we should carefully consider why we'd want to replace something that does the job.

To me even if the two algorithms do the job equivalently in this particular case, there is value in putting `picard` here if 1) we expect it to be better in the general case and 2) we want to get people to try it. This is a valuable educational / advertising spot.

Reply:
I agree these are valid reasons. Although I'd prefer a better example that really showcases `picard` as opposed to an example where FastICA is apparently just as good. Why not add such a dedicated example? Anyway, you can go ahead and include it in the existing tutorial, I'm not strongly against it - just trying to argue from a user's perspective.

Reply:
> Why not add such a dedicated example?

Hmm... true, we could create a new example and point to it from this one (in this spot about which method to use), which keeps visibility high. @pierreablin you could translate the example from your other repo over here as a tutorial to convince people. Then if we ever decide to change the default from FastICA to picard, we change the method here. @pierreablin WDYT?